### PR TITLE
Sc apiserver metrics

### LINKED ISSUE #1719: Add apiserver related charts in to stackdriver monitoring dashboard.
**Description**

Our weekly and nightly clusters are running with stackdriver collector. We would like to use it for monitoring api servers. During installing stackdriver we encounter issues in setting up metrics mentioned in acceptance criteria. At present no metrics about apiserver are delivered to stackdriver. Error about unknown resource type is written in collector logs for metric with job="apiserver".

Stackdriver should provide same metrics as grafana dashboard created in https://github.com/kyma-project/kyma/issues/6035.

- Chart with communication with API server grouped by component
- Chart with requests grouped by type (same as in Kubernetes Control Plane Status dashboard) with the ability to filter per component.
- Table with TOP 10 components

**Reasons**

We need to monitor API servers performance in long running clusters to react on changes possibly damaging cluster performance.

### PR DESCRIPTION
**Pull request by dekiel**
**Description**

Kyma long lasting integration script now create secret from config file not just apply precreated secret.

Changes proposed in this pull request:

- Corrected stackdriver sidecar regex filters.
- Added documentation about implemented changes.
- Changed creating additional scrape config secret from config file.

Closes #1719

### CODE REVIEW

File: prow/scripts/resources/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+## Overview
+
+The folder Contains files which are directly used by prow pipeline scripts.
+
+## Content
+
+```
+├── kube-dns-stub-domains-patch.yaml                          # Enable stubdomain build.kyma-project.io and provide Google root DNS servers IPs.
+├── limitrange-patch.yaml                                     # Increase kyma-system namespace max memory request for containers.
+├── prometheus-operator-stackdriver-patch.yaml                # Inject stackdriver collector sidecar, set metrics filters, enable scraping stackdriver target.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
├── prometheus-operator-stackdriver-patch.yaml                # Injects the Stackdriver collector sidecar, sets metric filters, and enables scraping Stackdriver target.
```

Comment (klaudiagrz) [reply prow/scripts/resources/README.md]: Scraping?

File: prow/scripts/resources/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+└── prometheus-operator-additional-scrape-config.yaml         # Additionall scrape config for prometheus oeprator.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
└── prometheus-operator-additional-scrape-config.yaml         # Additionall scrape configuration for Prometheus operator.
```

Comment (klaudiagrz) [reply prow/scripts/resources/README.md]: Please explain scrape to me.

File: prow/scripts/resources/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+```
+
+## Prometheus operator additional scrape config
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
## Prometheus operator additional scrape configuration
```

Comment (klaudiagrz) [reply prow/scripts/resources/README.md]: Scrape?

File: prow/scripts/resources/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+Prometheus operator expect to have additional scrape config provided as one secret. This secret is appended to the Prometheus scrape config file as it is.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
Prometheus operator expects to have additional scrape configuration provided as a Secret. This Secret is appended to the Prometheus scrape config file.
```

File: prow/scripts/resources/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+Additional scrape config allow add scrape targets outside automatic scrape targets discovery mechanisms.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
Additional scrape config allows you to add scrape targets outside automatic scrape targets discovery mechanisms.
```

File: prow/scripts/resources/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+It's admin responsibility to provide syntactically correct scrape configuration.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
It is administrator's responsibility to provide syntactically correct scrape configuration.
```

File: prow/scripts/resources/README.md (hunk 1 above)
Comment (dekiel) [reply prow/scripts/resources/README.md]: Scraping is a word used by Prometheus documentation to describe process of collecting metric values from monitored components.

File: prow/scripts/resources/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+## Overview
+
+The folder Contains files which are directly used by prow pipeline scripts.
+
+## Content
+
+```
+├── kube-dns-stub-domains-patch.yaml                          # Enable stubdomain build.kyma-project.io and provide Google root DNS servers IPs.
```
[the remark this answers is not in the document]
Comment (dekiel) [reply prow/scripts/resources/README.md]: It's not a typo in stubdomain word.

File: prow/scripts/cluster-integration/README.md [hunk 8]
```diff
@@ -68,3 +68,11 @@ This service account must have GCP permissions equivalent to the following GCP r
 - DNS Administrator (`roles/dns.admin`)
 - Service Account User (`roles/iam.serviceAccountUser`)
 - Storage Admin (`roles/storage.admin`)
+
+### Stackdriver monitoring
+
+Long running clusters on GKE are using Stackdriver monitoring to expose some performance metrics scraped by Kyma Prometheus instance.
+To send metrics to Stackdriver,collector sidecar container is injected in to Prometheus pod.
+Add container enviroment variable to instruct script to provision cluster with Stackdriver collector.
+    - name: STACKDRIVER_COLLECTOR_SIDECAR_IMAGE_TAG
```
[the remark this answers is not in the document]
Comment (dekiel) [reply prow/scripts/cluster-integration/README.md]: Why you would like to remove this?
It's example of added container environment variable.

File: prow/scripts/cluster-integration/README.md [hunk 9]
```diff
@@ -68,3 +68,15 @@ This service account must have GCP permissions equivalent to the following GCP r
 - DNS Administrator (`roles/dns.admin`)
 - Service Account User (`roles/iam.serviceAccountUser`)
 - Storage Admin (`roles/storage.admin`)
+
+### Stackdriver Monitoring
+
+Long-running clusters on GKE use Stackdriver Monitoring to expose some performance metrics scraped by Kyma Prometheus instance.
```
Comment (klaudiagrz) [inline prow/scripts/cluster-integration/README.md]:
```suggestion
Long-running clusters on GKE use Stackdriver Monitoring to expose some performance metrics collected by Kyma Prometheus instance.
```

File: prow/scripts/cluster-integration/README.md [hunk 10]
```diff
@@ -68,3 +68,15 @@ This service account must have GCP permissions equivalent to the following GCP r
...
+To send metrics to Stackdriver, the collector sidecar container is injected into Prometheus Pod.
+Add the container's environment variable to instruct the script to provision cluster with the Stackdriver collector.
+Long running clusters on GKE are using Stackdriver monitoring to expose some performance metrics scraped by Kyma Prometheus instance.
```
Comment (klaudiagrz) [inline prow/scripts/cluster-integration/README.md]: This sentence is a repetition. We can delete it.

File: prow/scripts/cluster-integration/README.md [hunk 11]
```diff
@@ -68,3 +68,15 @@ This service account must have GCP permissions equivalent to the following GCP r
...
+To send metrics to Stackdriver,collector sidecar container is injected in to Prometheus pod.
```
Comment (klaudiagrz) [inline prow/scripts/cluster-integration/README.md]: This sentence is a repetition. We can delete it.

File: prow/scripts/cluster-integration/README.md [hunk 12]
```diff
@@ -68,3 +68,15 @@ This service account must have GCP permissions equivalent to the following GCP r
...
+Add container enviroment variable into prowjob specification to instruct script to provision cluster with Stackdriver collector.
```
Comment (klaudiagrz) [inline prow/scripts/cluster-integration/README.md]: This sentence is a repetition. We can delete it.

File: prow/scripts/cluster-integration/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (klaudiagrz) [reply prow/scripts/cluster-integration/README.md]: My intention was to put both variables in the code block, just as you did. It displays better now.

File: prow/scripts/resources/README.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
+## Overview
+
+The folder contains files that are directly used by Prow pipeline scripts.
+
+## Directory structure
+
+```
+├── kube-dns-stub-domains-patch.yaml                          # Enable stubdomain build.kyma-project.io and provides Google root DNS servers IPs.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
├── kube-dns-stub-domains-patch.yaml                          # Enables the build.kyma-project.io stubdomain and provides Google root DNS servers IPs.
```

File: prow/scripts/resources/README.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+├── limitrange-patch.yaml                                     # Increases the kyma-system Namespace maximum memory request for containers.
+├── prometheus-operator-stackdriver-patch.yaml                # Injects the Stackdriver collector sidecar, sets metric filters, and enables scraping Stackdriver target.
+└── prometheus-operator-additional-scrape-config.yaml         # Additionall scrape configuration for Prometheus operator.
+├── kube-dns-stub-domains-patch.yaml                          # Enables stubdomain build.kyma-project.io and provides Google root DNS servers IPs.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]: From this point, the structure is repeated. We can delete it.

File: prow/scripts/resources/README.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+
+The folder contains files that are directly used by Prow pipeline scripts.
+
+## Directory structure
+
+```
+├── kube-dns-stub-domains-patch.yaml                          # Enable stubdomain build.kyma-project.io and provides Google root DNS servers IPs.
+├── limitrange-patch.yaml                                     # Increases the kyma-system Namespace maximum memory request for containers.
+├── prometheus-operator-stackdriver-patch.yaml                # Injects the Stackdriver collector sidecar, sets metric filters, and enables scraping Stackdriver target.
+└── prometheus-operator-additional-scrape-config.yaml         # Additionall scrape configuration for Prometheus operator.
```
Comment (klaudiagrz) [inline prow/scripts/resources/README.md]:
```suggestion
└── prometheus-operator-additional-scrape-config.yaml         # Additional scrape configuration for Prometheus operator.
```

File: prow/scripts/resources/README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (dekiel) [reply prow/scripts/resources/README.md]: Other suggestions commited.

File: prow/scripts/cluster-integration/README.md (hunk 10 above)
Comment (dekiel) [reply prow/scripts/cluster-integration/README.md]: done

File: prow/scripts/cluster-integration/README.md (hunk 11 above)
Comment (dekiel) [reply prow/scripts/cluster-integration/README.md]: done

File: prow/scripts/cluster-integration/README.md (hunk 12 above)
Comment (dekiel) [reply prow/scripts/cluster-integration/README.md]: done

File: prow/scripts/resources/README.md (hunk 14 above)
Comment (dekiel) [reply prow/scripts/resources/README.md]: done

**Pull request merged**